etna/search/views.py

```diff
@@ -571,6 +576,73 @@ class WebsiteSearchView(LoginRequiredMixin, BucketsMixin, BaseFilteredSearchView
     template_name = "search/catalogue_search.html"
     title_base = "Catalogue results"
 
+    def add_insights_page_for_url(self, page: Page) -> None:
+        """
+        Finds the Insights page corresponding to the sourceUrl of a record, then adds that page to result of the same record.
+        Unmatched url is bypassed but logged.
+        """
+        insights_page_by_url = {
+            page.get_url(self.request): page
+            for page in InsightsPage.objects.live()
+            .defer("body")
+            .select_related("teaser_image")
```

Comment:
@TNA-Allan For a nice 'efficiency bump' here, how do you feel about limiting this queryset, so that only the required pages are fetched? This might help reduce unnecessary queries when the page is empty, for example. Filtering on `slug` might be an option (if those can be easily extracted from the URLs)?